### PR TITLE
Adding Extends Interfaces

### PR DESCRIPTION
**Pull request by jack-farley**
This PR allows interfaces defined in ontology as code to extend other interfaces.

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/osdk-ts](https://github.com/palantir/osdk-ts), @jack-farley! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: packages/maker/src/api/defineInterface.ts [hunk 1]
```diff
@@ -34,6 +34,7 @@ export function defineInterface(
       string,
       SharedPropertyType | PropertyTypeType
     >;
+    extends?: InterfaceType | InterfaceType[] | string | string[];
```
Comment (nathklei) [inline packages/maker/src/api/defineInterface.ts:37]: can you add/update test cases in https://github.com/palantir/osdk-ts/blob/main/packages/maker/src/api/overall.test.ts?

Comment (jack-farley) [reply packages/maker/src/api/defineInterface.ts:37]: Done

**Comment by nathklei** [comment]
@jack-farley did you mean to close instead of merge?

**Pull request closed without merging**